warcio/limitreader.py

```diff
@@ -67,3 +74,135 @@ def wrap_stream(stream, content_length):
 
         return stream
 
+
+# ============================================================================
+class DigestVerifyingReader(LimitReader):
```

Comment:
I think it'd be good to move this class (and supporting functions, possible sublcasses, etc..) to a separate file, eg. digestverifyingreader.py, to keep limitreader.py small and self-contained for what it does